src/Notepads/Controls/TextEditor/TextEditorCore.DateTime.cs

```diff
@@ -6,11 +6,28 @@
 
     public partial class TextEditorCore
     {
+        private bool _hasAddedLogEntry = false;
+
         private void InsertDateTimeString()
         {
             var dateStr = DateTime.Now.ToString(CultureInfo.CurrentCulture);
             Document.Selection.SetText(TextSetOptions.None, dateStr);
             Document.Selection.StartPosition = Document.Selection.EndPosition;
         }
+
+        public void TryInsertNewLogEntry()
+        {
+            var docText = GetText();
+            if (!docText.StartsWith(".LOG") || _hasAddedLogEntry) return;
```

Comment:
check _hasAddedLogEntry first in the first line so you can skip getting text in some cases.

Reply:
Done.